src/components/ComputeModal.js

```diff
@@ -1377,15 +1381,15 @@ export const ComputeModalBase = ({ onDismiss, onSuccess, runtimes, persistentDis
       options: [
         {
           label: 'TERRA-MAINTAINED JUPYTER ENVIRONMENTS',
-          options: getImages(({ isCommunity, isRStudio }) => (!isCommunity && !isRStudio))
+          options: getImages(({ isCommunity, id }) => (!isCommunity && !(getToolForImage(id) === tools.RStudio.label)))
         },
         {
           label: 'COMMUNITY-MAINTAINED JUPYTER ENVIRONMENTS (verified partners)',
           options: getImages(_.get(['isCommunity']))
         },
         {
           label: 'COMMUNITY-MAINTAINED RSTUDIO ENVIRONMENTS (verified partners)',
-          options: getImages(_.get(['isRStudio']))
+          options: getImages((image) => getToolForImage(image.id) === tools.RStudio.label)
```

Comment:
Can be
```suggestion
          options: getImages(image => getToolForImage(image.id) === tools.RStudio.label)
```
since the function input takes a single input (i.e. `image`)